WebRoot/js/zimbraAdmin/servers/view/ZaNewVolumeXWizard.js

```diff
@@ -0,0 +1,1062 @@
+ZaNewVolumeXWizard.helpURL =
+    location.pathname + ZaUtil.HELP_URL + "managing_accounts/create_an_account.htm?locid=" + AjxEnv.DEFAULT_LOCALE;
+ZaNewVolumeXWizard.prototype.handleXFormChange = function () {
+    if (this._localXForm.hasErrors()) {
+        this._button[DwtWizardDialog.FINISH_BUTTON].setEnabled(false);
+    } else {
+        // Validate new volume step fields
+        if (this._containedObject[ZaModel.currentStep] !== ZaNewVolumeXWizard.GENERAL_STEP) {
+            var areCommonVolumeFieldsEmpty = this.hasEmptyFields([ZaServer.A_VolumeName, ZaServer.A_VolumeType]);
+            var areInternalVolumeFieldsEmpty = this.hasEmptyFields([
+                ZaServer.A_VolumeRootPath,
+                ZaServer.A_VolumeCompressionThreshold,
+            ]);
+            var areExternalVolumeFieldsEmpty = this.hasEmptyFields([
+                ZaServer.A_VolumePrefix,
+                ZaServer.A_CompatibleS3Bucket,
+            ]);
+
+            if (
+                areCommonVolumeFieldsEmpty ||
+                (this._containedObject[ZaModel.currentStep] === ZaNewVolumeXWizard.NEW_INTERNAL_VOLUME &&
+                    areInternalVolumeFieldsEmpty) ||
+                (this._containedObject[ZaModel.currentStep] !== ZaNewVolumeXWizard.NEW_INTERNAL_VOLUME &&
+                    areExternalVolumeFieldsEmpty)
+            ) {
+                this._button[DwtWizardDialog.FINISH_BUTTON].setEnabled(false);
+            } else {
+                this._button[DwtWizardDialog.FINISH_BUTTON].setEnabled(true);
+            }
+        }
+
+        // Validate new bucket step fields
+        if (this._containedObject[ZaModel.currentStep] === ZaNewVolumeXWizard.NEW_S3_BUCKET) {
+            this._button[DwtWizardDialog.NEXT_BUTTON].setEnabled(false);
+        }
+    }
+};
+
+ZaNewVolumeXWizard.prototype.hasEmptyFields = function (fieldArray) {
+    var result = false;
+    for (var i in fieldArray) {
+        if (!this._containedObject[fieldArray[i]] || this._containedObject[fieldArray[i]] === "") {
+            result = true;
+            break;
+        }
+    }
+    return result;
+};
+
+/**
+ * Overwritten methods that control wizard's flow (open, go next,go previous, finish)
+ **/
+ZaNewVolumeXWizard.prototype.popup = function (loc) {
+    ZaXWizardDialog.prototype.popup.call(this, loc);
+    this._button[DwtWizardDialog.NEXT_BUTTON].setEnabled(true);
+    this._button[DwtWizardDialog.FINISH_BUTTON].setEnabled(false);
+    this._button[DwtWizardDialog.PREV_BUTTON].setEnabled(false);
+};
+
+ZaNewVolumeXWizard.prototype.goNext = function () {
+    if (this._containedObject[ZaModel.currentStep] === ZaNewVolumeXWizard.GENERAL_STEP) {
+        this.setDefaultValues();
+        this._button[DwtWizardDialog.PREV_BUTTON].setEnabled(true);
+        this._button[DwtWizardDialog.NEXT_BUTTON].setEnabled(false);
+
+        if (this._containedObject[ZaServer.A_VolumeStorageType] === "Internal") {
```

Comment:
Should we create constants for the strings we are using for comparison?

Reply:
done